### PR TITLE
Rodrigom/benchmarking support api v2

### PR DESCRIPTION
**Pull request by snova-rodrigom**
- Changed code to support api v2
- Adapted code to use URI as env variable
- Updated Readme to indicate streaming is only supported for now and api v2 is now supported
- Tested with api v2, COE and stand alone endpoints

### CODE REVIEW

File: benchmarking/src/performance_evaluation.py [hunk 1]
```diff
@@ -42,15 +44,16 @@ def __init__(
...
     ):
+        base_uri = os.environ.get("SAMBASTUDIO_BASE_URI")
+        
         self.model_name = model_name
         self.results_dir = results_dir
         self.num_workers = num_workers
         self.user_metadata = user_metadata
         self.llm_api = llm_api
-        self.generation_mode = generation_mode
+        self.generation_mode = "stream" if "stream" in base_uri else ""
```
Comment (snova-jorgep) [inline benchmarking/src/performance_evaluation.py:56]: same as comment above, is this expected to be in the base uri?

File: benchmarking/README.md [hunk 2]
```diff
@@ -68,17 +68,19 @@ git clone https://github.com/sambanova/ai-starter-kit.git
...
     - Assume you have an endpoint with the URL
-        "https://api-stage.sambanova.net/api/predict/generic/12345678-9abc-def0-1234-56789abcdef0/456789ab-cdef-0123-4567-89abcdef0123"
+        "https://api-stage.sambanova.net/api/predict/generic/stream/12345678-9abc-def0-1234-56789abcdef0/456789ab-cdef-0123-4567-89abcdef0123"
 
     - You can enter the following in the env file (with no spaces):
 
     ``` bash
         SAMBASTUDIO_BASE_URL="https://api-stage.sambanova.net"
-        SAMBASTUDIO_BASE_URI="api/predict/generic"
+        SAMBASTUDIO_BASE_URI="api/predict/generic/stream"
```
[the remark this answers is not in the document]
Comment (snova-rodrigom) [reply benchmarking/README.md:77]: I have removed it from the env variable and it's being added in code as other kits. Thanks for the suggestion

File: benchmarking/src/performance_evaluation.py (hunk 1 above)
Comment (snova-rodrigom) [reply benchmarking/src/performance_evaluation.py]: this is now a boolean value internally and as explained in previous comment it's being added in URI inside the code as other kits

File: benchmarking/src/llmperf/sambanova_client.py [hunk 3]
```diff
@@ -117,16 +109,22 @@ def _get_data(request_config: RequestConfig) -> dict:
...
-        data = {"instance": prompt, "params": json.loads(tuning_params)}
-    else:
-        data = {"instances": [prompt], "params": json.loads(tuning_params)}
+        }
+        tuning_params = json.dumps(tuning_params_dict)
+        
+        if request_config.mode == "stream":
+            data = {"instance": prompt, "params": json.loads(tuning_params)}
+        else:
+            data = {"instances": [prompt], "params": json.loads(tuning_params)}
```
[the remark this answers is not in the document]
Comment (snova-rodrigom) [reply benchmarking/src/llmperf/sambanova_client.py]: so far, we haven't required compatibility with `api/predict/nlp`. It's something that we can consider in future and will definitely add it to the backlog. Thanks for the observation!

File: benchmarking/README.md (hunk 2 above)
Comment (snova-jorgep) [inline benchmarking/README.md:77]: then could you please remove it from the readme as well

**Review by snova-jorgep — CHANGES_REQUESTED** [review]

File: benchmarking/README.md (hunk 2 above)
Comment (snova-rodrigom) [reply benchmarking/README.md]: good catch! thanks!

**Pull request merged**